extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorpvault/HashicorpVaultClient.java

```diff
@@ -0,0 +1,166 @@
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Objects;
+
+import static dev.failsafe.Failsafe.with;
+
+class HashicorpVaultClient {
+    static final String VAULT_DATA_ENTRY_NAME = "content";
+    private static final String VAULT_TOKEN_HEADER = "X-Vault-Token";
+    private static final String VAULT_REQUEST_HEADER = "X-Vault-Request";
+    private static final MediaType MEDIA_TYPE_APPLICATION_JSON = MediaType.get("application/json");
+    private static final String VAULT_API_VERSION = "v1";
+    private static final String VAULT_SECRET_PATH = "secret";
+    private static final String VAULT_SECRET_DATA_PATH = "data";
+    private static final String VAULT_SECRET_METADATA_PATH = "metadata";
+    private static final String CALL_UNSUCCESSFUL_ERROR_TEMPLATE = "[Hashicorp Vault] Call unsuccessful: %s";
+    @NotNull
+    private final HashicorpVaultClientConfig config;
+    @NotNull
+    private final OkHttpClient okHttpClient;
+    @NotNull
+    private final TypeManager typeManager;
+    @NotNull
+    private final RetryPolicy<Object> retryPolicy;
+
+    HashicorpVaultClient(@NotNull HashicorpVaultClientConfig config, @NotNull OkHttpClient okHttpClient, @NotNull TypeManager typeManager, @NotNull RetryPolicy<Object> retryPolicy) {
+        this.config = config;
+        this.okHttpClient = okHttpClient;
+        this.typeManager = typeManager;
+        this.retryPolicy = retryPolicy;
+    }
+
+    Result<String> getSecretValue(@NotNull String key) {
+        var requestUri = getSecretUrl(key, VAULT_SECRET_DATA_PATH);
+        var headers = getHeaders();
+        var request = new Request.Builder().url(requestUri).headers(headers).get().build();
+
+        try (var response = with(retryPolicy).get(() -> okHttpClient.newCall(request).execute())) {
+
+            if (response.isSuccessful()) {
+                if (response.code() == 404) {
+                    return Result.failure(
+                            String.format(CALL_UNSUCCESSFUL_ERROR_TEMPLATE, "Secret not found"));
+                }
+
+                var responseBody = response.body();
+                if (responseBody == null) {
+                    return Result.failure(String.format(CALL_UNSUCCESSFUL_ERROR_TEMPLATE, "Response body empty"));
+                }
+                var payload = typeManager.readValue(responseBody.string(), HashicorpVaultGetEntryResponsePayload.class);
+                var value = payload.getData().getData().get(VAULT_DATA_ENTRY_NAME);
+
+                return Result.success(value);
+            } else {
+                return Result.failure(String.format(CALL_UNSUCCESSFUL_ERROR_TEMPLATE, response.code()));
+            }
+
+        } catch (IOException e) {
+            return Result.failure(e.getMessage());
+        }
+    }
+
+    Result<HashicorpVaultCreateEntryResponsePayload> setSecret(
```

Comment:
I cannot see a reason for this to be package-private. Even if the class is package-private, the methods should either be `public` and be tested, or be `private` to indicate clearly what the intention was.

Reply:
These methods were tested in previous versions of these had unit tests. But it was decided that testing them via integration tests was enough. If this stance has changed, we can bring those tests back, but it would be great if the code owners could align their requirements here.

As to them not being private: If they were, we couldn't use them.

Reply:
I understand that the intention was to make methods package-private so that other classes have access to them, but my point was not to make them _private_, but make them _public_. 

Simply to make the intent clear to any future reader, that these methods should be called by collaborators.